meinberlin/apps/dashboard/views.py

```diff
@@ -52,3 +59,64 @@ def get_success_url(self):
         return reverse(
             'a4dashboard:newsletter-create',
             kwargs={'organisation_slug': self.organisation.slug})
+
+
+class ModuleBlueprintListView(a4dashboard_mixins.DashboardBaseMixin,
+                              generic.DetailView):
+    template_name = 'meinberlin_dashboard/module_blueprint_list.html'
+    permission_required = 'a4projects.add_project'
+    model = project_models.Project
+    slug_url_kwarg = 'project_slug'
+    menu_item = 'project'
+
+    @property
+    def blueprints(self):
+        return [
+            (key, data) for key, data in get_blueprints()
+            if key not in ['bplan', 'external-project']
+        ]
+
+
+class ModuleCreateView(a4dashboard_mixins.DashboardBaseMixin,
+                       a4dashboard_mixins.BlueprintMixin,
+                       SingleObjectMixin,
+                       generic.View):
+    permission_required = 'a4projects.add_project'
+    model = project_models.Project
+    slug_url_kwarg = 'project_slug'
+
+    def post(self, request, *args, **kwargs):
+        project = self.get_object()
+        module = module_models.Module(
+            name=self.blueprint.title,
+            weight=len(project.modules) + 1,
+            project=project,
+        )
+        module.save()
+
+        self._create_module_settings(module)
+        self._create_phases(module, self.blueprint.content)
+
+        return HttpResponseRedirect(self.get_next(module))
+
+    def _create_module_settings(self, module):
+        if self.blueprint.settings_model:
+            settings_model = apps.get_model(*self.blueprint.settings_model)
+            module_settings = settings_model(module=module)
+            module_settings.save()
+
+    def _create_phases(self, module, blueprint_phases):
+        for phase_content in blueprint_phases:
+            phase = phase_models.Phase(
+                type=phase_content.identifier,
+                name=phase_content.name,
+                description=phase_content.description,
+                weight=phase_content.weight,
+                module=module,
+            )
+            phase.save()
+
+    def get_next(self, module):
```

Comment:
i would prefer `get_success_url` which is normally used by us (due to the FormMixin)

Reply:
actually i take it back.this is not in the context of a module, so we have to pass it along. sorry